lib/class_kit/class_methods.rb

```diff
@@ -1,58 +1,55 @@
 module ClassKit
-  def attr_accessor_type(name, type: nil, collection_type: nil, allow_nil: true, default: nil, auto_init: false, meta: {})
+  def attr_accessor_type(name, type: nil, collection_type: nil, allow_nil: true, default: nil, auto_init: false,
+                         meta: {})
 
-    if !self.class_variable_defined?(:@@class_kit_attributes)
-      self.class_variable_set(:@@class_kit_attributes, [])
+    unless class_variable_defined?(:@@class_kit_attributes)
+      class_variable_set(:@@class_kit_attributes, [])
     end
 
-    if !self.class_variable_defined?(:@@is_class_kit_class)
-      self.class_variable_set(:@@is_class_kit_class, true)
-    end
+    attributes = class_variable_get(:@@class_kit_attributes)
 
-    attributes = self.class_variable_get(:@@class_kit_attributes)
-    attributes.push({ name: name, type: type, collection_type: collection_type, allow_nil: allow_nil, default: default, auto_init: auto_init, meta: meta })
-    self.class_variable_set(:@@class_kit_attributes, attributes)
+    attributes.push({ name: name, type: type, collection_type: collection_type, allow_nil: allow_nil,
+                                  default: default, auto_init: auto_init, meta: meta })
 
-    self.class_eval do
+    class_eval do
       define_method name do
-
         cka = self.class.class_variable_get(:@@class_kit_attributes).detect { |i| i[:name] == name }
```

Comment:
This comes with a performance penalty, the array has to be traversed each time this method is called until the correct hash has been found. Using an Hash for `@@class_kit_attributes` could fix that, also it would help with the case when `name` is not unique.